src/common/gpx.c

```diff
@@ -173,13 +259,78 @@ gboolean dt_gpx_get_location(struct dt_gpx_t *gpx, GDateTime *timestamp, dt_imag
       return FALSE;
     }
 
+    dt_gpx_track_point_t *tp_next = (dt_gpx_track_point_t *)item->next->data;
     /* check if timestamp is within current and next trackpoint */
-    const gint cmp_n = g_date_time_compare(timestamp, ((dt_gpx_track_point_t *)item->next->data)->time);
+    const gint cmp_n = g_date_time_compare(timestamp, tp_next->time);
     if((cmp >= 0) && (item->next && cmp_n <= 0))
     {
-      geoloc->longitude = tp->longitude;
-      geoloc->latitude = tp->latitude;
-      geoloc->elevation = tp->elevation;
+      GTimeSpan seg_diff = g_date_time_difference(tp_next->time, tp->time);
+      GTimeSpan diff = g_date_time_difference(timestamp, tp->time);
+      if (seg_diff == 0)
```

Comment:
should be (seg_diff == 0 || diff == 0)
seg_diff == 0 is unlikely

Reply:
done!

even if unlikely, `seg_diff == 0` would produce a crash in line 289 (division by zero)

Reply:
Agreed but have you already got a null length segment ?

Reply:
I make gpx traces with osmand, and it has the options to either produce a new point even if no movement or not produce it. In the first case I suppose the null length segments are produced. Anyway I wouldn't expose darktable to the risk a division by zero crash.

Reply:
Thanks. Good to know.